### PR TITLE
Configuration and Multiple Networks

### LINKED ISSUE #4: Switching between modes of operation
The game/simulation is assigned in two phases. During the first "testnet" phase students are expected to get themselves familiarized with the API and authentication. The system is restarted and the "realnet" phase begins. Ideally, these two phases have different rules so there's something to look forward to. The modes should also be changed between different semesters (different offerings of the course). 

Currently, this is handled manually using git branches or commenting/uncommenting and copy & pasting. A more robust solution might look like a command line or environment variable that sets the rules of the game according to the mode. Codebase probably requires heavy refactoring to allow this as well.

### PR DESCRIPTION
**Pull request by necrashter**
Closes #4.

## Features
- Constants are moved to a `Config` struct, which has docstrings for its fields.
- Current fields in `Config`:
  - Coin constants (transaction limits, rewards, block count, etc.)
  - Preapproved user list (name of the CSV file)
  - The number of hexadecimal zeros a hash value should start with
  - URL prefix (e.g. serve this network at `/testnet/api-command` instead of `/api-command`)
  - The name of the network
- `Config` struct can be read from `.yaml` files.
- While saving the DB, the name of the network is taken into account.
  - `users/network_name/...` and `blocks/network_name/...`.
- Config file is `config.yaml` by default, can be overridden by giving command line arguments.
- **Multiple networks can run at the same time** by giving them as command line arguments, as long as they have different names and URL prefixes.
  - Test network can continue running in `/testnet` endpoint while the main network runs at `/`.
- Users can get the current configuration as JSON from `GET /config` endpoint.

## Other updates
- Instructions to run locally in `README`.
- Previous code read the users, but overwrote them during DB construction. Fixed.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "gradecoin"
 version = "0.2.0"
-authors = ["Yigit Sever <yigit@ceng.metu.edu.tr>"]
```
Comment (yigitsever) [inline Cargo.toml:4]: We should Alperen Keles too

**Review by yigitsever — CHANGES_REQUESTED** [review]

**Comment by yigitsever** [comment]
Thank you so much for your work (and solving the overriding block/user bugs, those were driving me crazy this morning), looks amazing

File: Cargo.toml (hunk 1 above)
Comment (necrashter) [reply Cargo.toml:4]: I added him and also signed my commits after the PR is merged. What should I do?

Comment (yigitsever) [reply Cargo.toml:4]: ah sorry it was something trivial so I quickly did it as you've already done plenty :) please grab the main branch if you want to contribute further

**Pull request merged**